hasher-matcher-actioner/hmalib/lambdas/ddb_stream_counter.py

```diff
@@ -85,16 +85,32 @@ def update_increments_for_records(
 
             pk: str = record["dynamodb"]["Keys"]["PK"]["S"]
             sk: str = record["dynamodb"]["Keys"]["SK"]["S"]
+            # The raw stream values have a different schema than what is returned by boto3 meaning
+            # they can't be deserialized into one of the HMA datamodels using existing functionality.
+            inserted_record = record["dynamodb"]["NewImage"]
```

Comment:
What happens when the record does not have "NewImage" as a key in the sub-dict?
I worry this might error for count increments that aren't around images. Can you confirm match and hash records are still incremented correctly after this change is deployed?

Reply:
I'll preface by saying this is my first time working with Dynamo DB. My understanding is that since we are only handling "INSERT" events here, there should always be a "NewImage" field that contains the data that was added. It's still probably safer to check though? On that note, should I also be safer when accessing the fields within `inserted_record`?

Reply:
I could add something like
```python
if (content_type := inserted_record.get("ContentType", {}).get("S")) is not None:
    count_buffer.inc_parameterized(
        AggregateCount.PipelineNames.submits, "content_type", content_type
    )
```

Reply:
Good point, I was finally able to trace down the format in the docs.
https://docs.aws.amazon.com/amazondynamodb/latest/APIReference/API_streams_StreamRecord.html

It looks like it is possible for INSERT events not have NewImage if the ViewType is KEYS_ONLY, however given this is something we define in terraform and the later existing checks we have would break anyway. I think it is fine as is.

Reply:
However the checks you are consider for `inserted_record` sound good to me. As it's possible those change could be missed and it would be better if those errors only affected parameterized counts instead of erroring on all records from this innovation. Thanks!

Reply:
However the checks you are consider for inserted_record sound good to me. As it's possible those change could be missed and it would be better if those errors only affected parameterized counts instead of erroring on all records from this innovation. Thanks!